File: survival game/Assets/TreeController.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class TreeController : MonoBehaviour
{
public int treeHealth = 5;

public Transform logs;
public GameObject tree;

public int speed = 8;
public bool fell = false;

//public Vector3 position = Vector3(Random.Range(-1.0, 1.0), 0, Random.Range(-1.0, 1.0));

void Start()
{
	tree = this.gameObject;
	GetComponent<Rigidbody>().isKinematic = true;
}

void Update()
{
    if(treeHealth <= 0 && fell == false)
    {
        fell = true;
        GetComponent<Rigidbody>().isKinematic = false;
        GetComponent<Rigidbody>().AddForce(transform.forward * speed);
        DestroyTree();
    }
}

void DestroyTree()
{
    new WaitForSeconds(5);

    Instantiate(logs, tree.transform.position, Quaternion.identity);

    Destroy(tree);

}

}

File: survival game/Assets/Visuals/TreeController.js
#pragma strict

var treeHealth : int = 5;

var logs : Transform;
var coconut : Transform;
var tree : GameObject;

var speed : int = 8;

function Start()
{
	tree = this.gameObject;
	rigidbody.isKinematic = true;
}

function Update()
{
	if(treeHealth <= 0)
	{
		rigidbody.isKinematic = false;
		rigidbody.AddForce(transform.forward * speed);
		DestroyTree();
	}
}

function DestroyTree()
{
	yield WaitForSeconds(7);
	Destroy(tree);

	var position : Vector3 = Vector3(Random.Range(-1.0, 1.0), 0, Random.Range(-1.0, 1.0));
	Instantiate(logs, tree.transform.position + Vector3(0, 0, 0) + position, Quarternion.identity);
	Instantiate(logs, tree.transform.position + Vector3(2, 2, 0) + position, Quarternion.identity);
	Instantiate(logs, tree.transform.position + Vector3(5, 5, 0) + position, Quarternion.identity);
}